lib/src/emoji_picker_internal_utils.dart

```diff
@@ -1,128 +1,36 @@
 import 'package:flutter/services.dart';
 import 'package:shared_preferences/shared_preferences.dart';
-import 'emoji_lists.dart' as emoji_list;
 import 'recent_emoji.dart';
 
 /// Helper class that provides internal usage
 class EmojiPickerInternalUtils {
   // Establish communication with native
   static const _platform = MethodChannel('emoji_picker_flutter');
-  static const _emojiVersion = 'emoji_version';
-
-  /// Returns true when local emoji list is outdated
-  Future<bool> isEmojiUpdateAvailable() async {
-    final prefs = await SharedPreferences.getInstance();
-    var emojiVersion = prefs.getInt(_emojiVersion) ?? 0;
-    // != to support downgrading of emoji_picker versions
-    return emoji_list.version != emojiVersion;
-  }
-
-  /// Updates local emoji version with current version
-  Future updateEmojiVersion() async {
-    final prefs = await SharedPreferences.getInstance();
-    prefs.setInt(_emojiVersion, emoji_list.version);
-  }
-
-  /// Restore locally cached emoji
-  Future<Map<String, String>?> _restoreFilteredEmojis(String title) async {
-    final prefs = await SharedPreferences.getInstance();
-    var emojiJson = prefs.getString(title);
-    if (emojiJson == null) {
-      return null;
-    }
-    var emojis =
-        Map<String, String>.from(jsonDecode(emojiJson) as Map<String, dynamic>);
-    return emojis;
-  }
 
   // Get available emoji for given category title
-  Future<Map<String, String>> _getAvailableEmojis(Map<String, String> map,
-      {required String title}) async {
-    Map<String, String>? newMap;
-
-    // Get Emojis cached locally if available
-    if (await isEmojiUpdateAvailable() == false) {
-      newMap = await _restoreFilteredEmojis(title);
-    }
-
-    if (newMap == null) {
-      // Check if emoji is available on this platform
-      newMap = await _getPlatformAvailableEmoji(map);
-      // Save available Emojis to local storage for faster loading next time
-      if (newMap != null) {
-        await _cacheFilteredEmojis(title, newMap);
-      }
-    }
-
-    return newMap ?? {};
-  }
-
-  /// Returns map of all the available category emojis
-  Future<Map<Category, Map<String, String>>> getAvailableCategoryEmoji() async {
-    final allCategoryEmoji = Map.fromIterables([
-      Category.SMILEYS,
-      Category.ANIMALS,
-      Category.FOODS,
-      Category.ACTIVITIES,
-      Category.TRAVEL,
-      Category.OBJECTS,
-      Category.SYMBOLS,
-      Category.FLAGS
-    ], [
-      emoji_list.smileys,
-      emoji_list.animals,
-      emoji_list.foods,
-      emoji_list.activities,
-      emoji_list.travel,
-      emoji_list.objects,
-      emoji_list.symbols,
-      emoji_list.flags,
+  Future<CategoryEmoji> _getAvailableEmojis(CategoryEmoji category) async {
+    var available = (await _platform.invokeListMethod<bool>(
+        'getSupportedEmojis', {
+      'source': category.emoji.map((e) => e.emoji).toList(growable: false)
+    }))!;
```

Comment:
Since you were aiming for performance improvements I'm surprise that you removed the local caching mechanism here. Are you sure that wouldn't be beneficial? But I understand this comes with the price of more complexity and to check for when `emoji-version` and `Config` are changed. 🤔 I will try to check with an older device to see the performance. This would be also possible to address in a separate PR when the performance is no issue for now.

Reply:
Indeed, my main problem with this is that if we keep this caching approach, we will have to force host apps care about emoji versioning if they provide a customized set (will have to explain it and somehow force it in config). Reading 60kb from SharedPreferences, I suppose, is also not free, but to my surprise it was fast. In the end, I'm planning to disable compatibility check in my project anyway, so I don't mind bringing this back.
My performance complains were mainly about scrolling, I don't think initialization delay will be noticeable in majority use cases. BTW, right now for me initialization is faster with recents `off` and compatibility check `on` than the other way around, because, as I've mentioned, SharedPreferences initialization is expensive.

Reply:
I did some testing with old devices and I couldn't find any issues with loading the emoji's directly without local caching. Pretty impressive - I think it's fine like this 👍